var/spack/repos/builtin/packages/neuron/package.py

```diff
@@ -0,0 +1,240 @@
+    and networks of neurons. NEURON models individual neurons via
+    the use of sections that are automatically subdivided into individual
+    compartments, instead of requiring the user to manually create
+    compartments. The primary scripting language is hoc but a Python
+    interface is also available."""
+
+    homepage = "https://www.neuron.yale.edu/"
+    url      = "http://www.neuron.yale.edu/ftp/neuron/versions/v7.4/nrn-7.4.tar.gz"
+    github   = "https://github.com/nrnhines/nrn"
+
+    version('7.4', '2c0bbee8a9e55d60fa26336f4ab7acbf')
+    version('7.3', '993e539cb8bf102ca52e9fefd644ab61')
+    version('7.2', '5486709b6366add932e3a6d141c4f7ad')
+    version('develop', git=github, preferred=True)
+
+    variant('mpi',           default=True,  description='Enable MPI parallelism')
+    variant('python',        default=True,  description='Enable python')
+    variant('static',        default=True,  description='Build static libraries')
+    variant('cross-compile', default=False, description='Build for cross-compile environment')
+    variant('multisend',     default=True,  description="Enable multi-send spike exchange")
+    variant('rx3d',          default=False, description="Enable cython translated 3-d rxd")
+
+    depends_on('automake',   type='build')
+    depends_on('autoconf',   type='build')
+    depends_on('libtool',    type='build')
+    depends_on('pkg-config', type='build')
+    depends_on('python@2.6:', when='+python')
+    depends_on('mpi', when='+mpi')
+
+    # on bg-q mpi needs to be enabled : todo test this
+    conflicts('~mpi', when='platform=bgq')
+
+    # pgi compiler can't build static libraries
+    conflicts('%pgi', when='+static')
+
+    def url_for_version(self, version):
+        url = "http://www.neuron.yale.edu/ftp/neuron/versions/v{0}/nrn-{0}.tar.gz"
+        return url.format(version, version)
+
+    def patch(self):
+        # aclocal need complete include path especially on os x
+        pkgconf_inc = '-I %s/share/aclocal/' % (self.spec['pkg-config'].prefix)
+        libtool_inc = '-I %s/share/aclocal/' % (self.spec['libtool'].prefix)
+        newpath = 'aclocal -I m4 %s %s' % (pkgconf_inc, libtool_inc)
+        filter_file(r'aclocal -I m4', r'%s' % newpath, "build.sh")
+
+    def get_arch_options(self, spec):
+        options = []
+
+        if 'bgq' in self.spec.architecture:
+            options.extend(['--enable-bluegeneQ',
+                            '--host=powerpc64',
+                            '--without-memacs'])
+
+        if 'cray' in self.spec.architecture:
+            options.extend(['--without-memacs',
+                            '--without-nmodl',
+                            'cross_compiling=yes'])
+
+            # TODO: on cray systems we get an error while linking even if
+            # we use cc wrapper. Hence add explicitly add mpi library
+            if spec.satisfies('+mpi'):
+                options.append('LIBS=-lmpich')
```

Comment:
Does this work for all MPI libraries? Try `spec['mpi'].libs.link_flags`. See http://spack.readthedocs.io/en/latest/llnl.util.html#llnl.util.filesystem.LibraryList.link_flags for details.